scapy/asn1/ber.py

```diff
@@ -337,6 +337,7 @@ class BERcodec_STRING(BERcodec_Object):
     tag = ASN1_Class_UNIVERSAL.STRING
     @classmethod
     def enc(cls,s):
+        s = raw(s)
         return chb(hash(cls.tag))+BER_len_enc(len(s))+raw(s) # Be sure we are encoding bytes
```

Comment:
As I said previously, this raw may be removed

Reply:
In that case you are allowed to change the spaces of this line (and make it better)

Reply:
This line s = raw(s) is used to fix #988, when using `ASN1_STRING(RandString())` s is RangeString, this will cause length encode incorrect. because len(s) and raw(s) will call `RandString._fix()` twice and got two different string.

Reply:
Oh, you mean line 341?

Reply:
Yes